File: backend/Services/Parsers/FoodGuessrParser.cs
using System.Text.RegularExpressions;
using backend.Models;

namespace backend.Services.Parsers
{
    public class FoodGuessrParser : BasicScoreResultParser
    {
        private ILogger<FoodGuessrParser> _logger;

        public FoodGuessrParser(ILogger<FoodGuessrParser> logger) : base(logger)
        {
            _logger = logger;
        }

        public override bool CountWinner => true;
        public override string GameName => "FoodGuessr";
        public override bool GolfScoring => false;
        public override string? HelpText => null;
        protected override Regex Parser => new Regex($@"{GameName} .+?Total score: (?<{ScoreGroup}>[\d|,|.]+)", RegexOptions.Singleline);
        protected override string? ExtraContent => null;
        public override string Url => "https://foodguessr.com";

        protected override string GetCleanResult(string result, Match parserResults)
        {
            var index = result.IndexOf("Can you beat my score? New game daily!");
            if (index > -1) {
                return result.Substring(0, index).Trim();
            }

            return result;
        }

        protected override DailyResult SetScore(DailyResult dailyResult, Match parserResults)
        {
            if (!parserResults.Groups.ContainsKey(ScoreGroup))
            {
                return dailyResult;
            }

            var scoreString = parserResults.Groups[ScoreGroup].Value
                .Replace(",", string.Empty)
                .Replace(".", string.Empty);
            if (int.TryParse(scoreString, out var score))
            {
                dailyResult.Score = score;
            }

            return dailyResult;
        }
    }
}

File: lambdas/src/parsers/foodguessr-parser.ts
import { DailyResult } from '../models/daily-result';
import { ResultParser, getGroup, hasGroup } from './base/result-parser';

export class FoodGuessrParser extends ResultParser {
  readonly countWinner = true;
  readonly gameName = 'FoodGuessr';
  readonly golfScoring = false;
  readonly helpText = null;
  readonly url = 'https://foodguessr.com';
  
  // Matches "FoodGuessr - 04 Nov 2024 GMT...Total score: 9,000 / 15,000"
  // Note: C# uses [\d|,|.] which incorrectly includes literal pipes - fixed here to [\d,.]
  protected readonly parser = new RegExp(
    `${this.gameName} .+?Total score: (?<score>[\\d,\\.]+)`,
    's'
  );
  
  protected getCleanResult(result: string, match: RegExpMatchArray): string {
    // C# only looks for "Can you beat my score? New game daily!" - NOT "Can you match"
    const index = result.indexOf('Can you beat my score? New game daily!');
    if (index > -1) {
      return result.substring(0, index).trim();
    }
    return result;
  }
  
  getScoreValue(dailyResult: DailyResult): string | null {
    return dailyResult.score?.toString() ?? null;
  }
  
  protected setScore(dailyResult: DailyResult, match: RegExpMatchArray): DailyResult {
    if (hasGroup(match, 'score')) {
      const scoreStr = getGroup(match, 'score')?.replace(/,/g, '').replace(/\./g, '');
      if (scoreStr) {
        const score = parseInt(scoreStr, 10);
        if (!isNaN(score)) {
          dailyResult.score = score;
        }
      }
    }
    return dailyResult;
  }
}
